### DISCUSSION TITLE
How to reduce object storage consumption?

### DISCUSSION THREAD

**Discussion by reinhart1010** [misskey-dev/misskey#9309] [Q&A]
Hello, we are utilizing object storage (S3 etc.) for our Misskey instance, and we've seen a spike in our bucket size from zero to 6.5 GBs over the course of 2 months.

In order to maintain our server operational costs as stable as possible, is it possible for us to automatically remove images from notes which are hosted from other peers but rarely requested by our users (some sort of cache invalidation)? We are fine if images are served directly from the original peer (aka. hotlinking) as they often disable CORS policy as well for the sake of federation. And even Pleroma serves images right from the original peer's source.

<img width="1405" alt="image" src="https://user-images.githubusercontent.com/17312341/206916250-7b8c50e0-9eab-47d5-a402-1f61844150a5.png">

Note that we are using Cloudflare's R2 which is known to not having a proper automated file deletion support. It has to be done manually by calling DeleteObject commands.

**Comment by ThatOneCalculator** [comment]
You can go to Control Panel > General > Files > turn "cache remote files" off

But for object storage, 6.5GB is basically nothing, prices are usually ~$5 per *terabyte*

**Comment by tassoman** [comment]
If you delete storage items from outside, does MK try to re-cache them on demand?
